genienlp/arguments.py

```diff
@@ -94,6 +94,10 @@ def parse_argv(parser):
                         help='whether to allow filtering on the validation sets')
     parser.add_argument('--val_batch_size', nargs='+', default=[256], type=int,
                         help='Batch size for validation corresponding to tasks in val tasks')
+    
+    parser.add_argument('--share_train_val_tasks', action='store_true', help='if true will override val_tasks so that'
+                                                                             ' train and val share the exact same task object')
```

Comment:
Is there any situation where we don't want this? I'd say just always do it, keep track of the task objects by name.

Reply:
It doesn't happen in our current experiments. But later on, if we want to try few-shot or zero-shot on tasks other than almond, we will need this option. I will change the default to be True but will keep the flag.

Reply:
No. Just make it so that tasks with the same name share the same object. There is no reason to have different objects for the same task, even if you have different lists of tasks.

Reply:
okay, that's reasonable.